### DISCUSSION TITLE
What is `aiMaterialProperty.mSemantic`?

### DISCUSSION THREAD

**Discussion by blair-ahlquist** [assimp/assimp#5266] [Q&A]
The code comments (https://github.com/assimp/assimp/blob/master/include/assimp/material.h#L632) read:

```
/** Textures: Specifies their exact usage semantic.
     * For non-texture properties, this member is always 0
     * (or, better-said, #aiTextureType_NONE).
     */
    unsigned int mSemantic;
```

What does this mean? I have an .obj file with the following properties (partial list, using C# wrapper):

```
...
Key: $tex.file, Type: PtiString, Index: 0, Semantic: 1, DataLength: 39, Data: BaseColor.png
Key: $tex.uvwsrc, Type: PtiInteger, Index: 0, Semantic: 1, DataLength: 4, Data: 0
Key: $tex.file, Type: PtiString, Index: 0, Semantic: 7, DataLength: 39, Data: Roughness.png
Key: $tex.uvwsrc, Type: PtiInteger, Index: 0, Semantic: 7, DataLength: 4, Data: 0
```

I can't tell how these textures are to be used, so I suspect the "semantic" has that info.

**Comment by janhsimon** [comment]
You're correct, it's an `aiTextureType`, fx. `aiTextureType_BASE_COLOR`. You can call `aiTextureTypeToString(prop->mSemantic)` to print the semantic in your example code above.

Hope this helps.
